Added: Favourite Engine Sounds

Alm0stEthical opened the pull request:
- I wanted to contribute by adding an option to add to favourites and removing favourites because, with _151 engine sounds_, it was repetitive to find/scroll all the way to get the exact sound I was looking for (After a server restart for example). I hope that my contribution improved efficiency!

# What I did
- I created a .json file that saves first by Discord ID the favourited engine sounds & if no Discord ID was found it will use the FiveM username/id. If that wasn't found either, all the data will be lost sadly 😥. I have no idea how to implement anti-data loss feature, I did this in .json so we didn't need to use databases, and keep this script as much standalone as possible.

# My future features I'd like to add
- Add a system to restrict sounds; let's say `npolchar`, maybe intergrate a method with FiveM's ACEs to restrict this sound to only that ACE that can access. For example group.soundpack1. Maybe that's a cool idea 💡. But that's it for me I hope you enjoyed contribution!

Alm0stEthical commented:
Oops, I forgot to add the video showcase because I was too excited. Here's a video that shows you how it works: 

https://dfuze.vip/github/pull-requests/gravxd/fivem-enginesound-menu/gravxd_enginesound-menu_showcase_pull_6.mp4

Gravxd commented:
I appreciate the ideas. I did have a planned update to put each engine sound behind a configurable permission so that you could technically do some cool stuff especially for rp servers - as for favourites, it's a good idea. I don't personally like doing json especially as a lot of servers use GITHUB/GITLAB for their development environments and editing json files on the server can be intensive if storing a lot of data. I'd prefer to use the easiest option - KVP. Completely client sided and the server doesn't need to worry about saving any data. Thanks for your input. Once i have some time, i'll look at making a new release with the included features.

Alm0stEthical commented:
Sure man, I know, JSON is kinda tricky I had a feeling but goodluck!

Alm0stEthical commented:
Do you want me to include KVP as well, and remove the JSON stuff? Thus, there's no need for you to reprogram? If, in any case, you're short on time

Gravxd commented:
If you want to, that'd be great - much more convenient and simpler for both the end user and the server using the resource.

Alm0stEthical commented:
Hi @Gravxd now it's only using KVP's and I removed all JSON related stuff to saving/loading and fixed an issue where the menu kept closing whilst the `CloseOnSelect` was set to `false`. Here is an updated preview: https://dfuze.vip/github/pull-requests/gravxd/fivem-enginesound-menu/gravxd_enginesound-menu_showcase_pull_6_video_2.mp4

Also would you like for me to add a check if you're in an airplane / chopper / boat so the menu doesn't open

Gravxd commented:
Thanks for your contributions, they look awesome. Sorry for the delay in review, have been busy.

Alm0stEthical commented:
No worries!
